storage/src/main/scala/io/casperlabs/storage/util/DoobieCodecs.scala

```
@@ -17,7 +17,14 @@ trait DoobieCodecs {
     Meta[Array[Byte]].imap(d => PublicKey(ByteString.copyFrom(d)))(_.toByteArray)
 
   protected implicit val readDeploy: Read[Deploy] =
-    Read[Array[Byte]].map(Deploy.parseFrom)
+    Read[(Array[Byte], Option[Array[Byte]])].map {
+      case (deploySummary, maybeDeployBody) =>
+        val deploy = Deploy.parseFrom(deploySummary)
+        val maybeBody = maybeDeployBody
+          .filterNot(_.isEmpty)
+          .map(Deploy.Body.parseFrom) orElse deploy.body
+        maybeBody.map(deploy.withBody).getOrElse(deploy)
```

Won't this silently drop an error during parsing of the deploy body and produce a deploy without a body? This doesn't make sense right? A deploy without a body.

---

`parseFrom` is not silent when it fails, it throws something. 
The `filterNot(_.isEmpty)` part is for the migrated rows which got their default values from `default x'';`, these records can have their bodies in the `summary` from which `deploy` was parsed.

---

`pareFrom` (which delegates to `$type.mergeFrom`) is very flexible. It throws only in extreme cases. Can we not migrate fields so that we don't need `filterNot(_.isEmpty)` ?

---

It would require Java migrations, a bit of an overkill before testnet.

---

Then, since we restart the network anyway. Why not just do it? I think keeping the code clean and simple is more important in this case.

---

Because this was made for the hackathon where we wanted to ask Tom to restart the nodes without losing the state. 

Unfortunately with SQLite we don't have a lot of good options: I have to add a default value, otherwise I'd have to recreate the whole table an all of its indexes to just add a `not null` constraint, which I think is another overkill, whereas this was a clean backwards compatible migration.

I agree with keeping the code clean and reflecting the possibilites though.

---

I thought that now, after the hackathon, we can just modify the `CREATE TABLE …` scripts.

---

I think that would teach us bad habits, all the files are named after the date and the ticket. If you go and change the definition of an earlier table, a following migration may not even work any more.